### PR TITLE
Fix SHA, SHAKE, and KECCAK ASM flag passing

### PR DESCRIPTION
**Pull request by juergenchrist**
Flags for ASM implementations of SHA, SHAKE, and KECCAK were only passed to
the FIPS provider and not to the default or legacy provider.  This left some
potential for optimization.  Pass the correct flags also to these providers.

Signed-off-by: Juergen Christ <jchrist@linux.ibm.com>

### CODE REVIEW

**Comment by p-steuer** [comment]
Thanks Juergen. The bug is also in 3.0, right ?

File: crypto/sha/build.info [hunk 1]
```diff
@@ -82,6 +82,14 @@ SOURCE[../../providers/libfips.a]= $COMMON
 # need to be applied to all affected libraries and modules.
 DEFINE[../../libcrypto]=$SHA1DEF $KECCAK1600DEF
 DEFINE[../../providers/libfips.a]=$SHA1DEF $KECCAK1600DEF
+DEFINE[../../providers/libdefault.a]=$SHA1DEF $KECCAK1600DEF
+# We only need to include the SHA1DEF and KECCAK1600DEF stuff in the
+# legacy provider when it's a separate module and it's dynamically
+# linked with libcrypto.  Otherwise, it already gets everything that
+# the static libcrypto.a has, and doesn't need it added again.
+IF[{- !$disabled{module} && !$disabled{shared} -}]
+  DEFINE[../providers/liblegacy.a]=$SHA1DEF $KECCAK1600DEF
```
Comment (slontis) [inline crypto/sha/build.info:91]: Why is this needed for the legacy provider? This only contains older algorithms

Comment (t8m) [reply crypto/sha/build.info:91]: It might be useful for future if we for example move SHA1 to the legacy provider at some point.

**Review by t8m — APPROVED** [review]

**Review by p-steuer — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks.
